docs/src/pages/components/autocomplete/Virtualize.js

```diff
@@ -3,50 +3,58 @@ import PropTypes from 'prop-types';
 import TextField from '@material-ui/core/TextField';
 import Autocomplete from '@material-ui/lab/Autocomplete';
 import useMediaQuery from '@material-ui/core/useMediaQuery';
+import ListSubheader from '@material-ui/core/ListSubheader';
 import { useTheme, makeStyles } from '@material-ui/core/styles';
-import { FixedSizeList } from 'react-window';
+import { VariableSizeList } from 'react-window';
+import { Typography } from '@material-ui/core';
 
 function renderRow(props) {
   const { data, index, style } = props;
-
-  return React.cloneElement(data[index], {
-    style: {
-      overflow: 'hidden',
-      textOverflow: 'ellipsis',
-      whiteSpace: 'nowrap',
-      display: 'block',
-      ...style,
-    },
-  });
+  return React.cloneElement(data[index], { style });
 }
 
 // Adapter for react-window
 const ListboxComponent = React.forwardRef(function ListboxComponent(props, ref) {
   const { children, ...other } = props;
+  const itemData = React.Children.toArray(children);
   const theme = useTheme();
-  const smUp = useMediaQuery(theme.breakpoints.up('sm'));
-  const itemCount = Array.isArray(children) ? children.length : 0;
+  const smUp = useMediaQuery(theme.breakpoints.up('sm'), { noSsr: true });
```

Comment:
Won't that create a hydration error?

Reply:
I didn't get hydration errors. I did see rendering problems with next.js 9 when this was disabled. It would miscalculate the height and render about 100px too high.

Reply:
The hydration warning is shallow by Next v8. Regarding the rendering issue, I think that it misses a key on react window, we need to invalid his cache.

Reply:
I got two renders, one with `smUp = false` and then one with `smUp = true`. The listbox would render first with items of 48px, then with items of 36 pixels. If it was positioned above the textfield it would float about 100px too high. By using `{ noSsr: true }` I only see one render happening.

![mui](https://user-images.githubusercontent.com/2109932/70478683-e4fa7280-1adb-11ea-87f2-3a0cd59c4491.gif)

Your codesandbox had `key={itemCount}` but that doesn't work.

Also, why would this give hydration problems? The menu is closed, is it still rendered?

Reply:
Ok, so we need to update the key?
```jsx
key={`${itemCount}-${itemSize}`}
```

https://codesandbox.io/s/virtualized-grouped-attempt-wm0hi

Reply:
> Also, why would this give hydration problems? The menu is closed, is it still rendered?

Oh right, this sounds good then. I'm really uncomfortable with this Next.js v8 bug we are dragging around.

Reply:
👍